kyaml/resid/gvk.go

```diff
@@ -157,7 +157,7 @@ func (x Gvk) Equals(o Gvk) bool {
 // a Service should come before things that refer to it.
 // Namespace should be first.
 // In some cases order just specified to provide determinism.
-var orderFirst = []string{
+var OrderFirst = []string{
```

Comment:
I'm not a big fan of exposing these constants (they're an implementation detail that we are unable to change, as discussed in the issues that led to this PR), or of having two separate implementations of GVK sorting (the new one in the transformer and the sort on GVK itself, which afaik exists for the purposes of the legacy sort).

A pattern that we are using in the newer builtins is to create a "filter" containing the vast majority of the logic, and have the plugin simply wrap that. I'm wondering if we could do something similar here, and ideally use that approach to both keep these constants private and avoid reimplementing the sort. Normally, the filters go in `api/filters`, but that wouldn't work since one of the callers is `kyaml/resid`.  Perhaps `kyaml/kio/filters`? @natasha41575 wdyt / any better ideas?

Reply:
It seems to me that GVK sorting should only be contained in the legacysortorder transformer.
I was surprised to see that `gvk.IsLessThan` is used in a couple of places, namely:
- `namebackreferences`
- `fieldspec`

May I ask why? They don't seem to be useful for anything in the places they're used. The merging functionality used in these places seems to only need equality.

So I would propose:
- Duplicating the relevant code for GVK sorting in the sortordertransformer.
- Remove the code from `gvk.go` in a subsequent PR (the ordering and `IsLessThan` stuff).

Have I missed anything? It seems to me that sorting config should be encapsulated by the ordering plugin only, why does anyone else need to know about it?

Reply:
In name back references, it seems to be used exclusively by:
```go
// sortFields provides determinism in logging, tests, etc.
func (t *TransformerConfig) sortFields()
```

That is called in the code path where we load transformer configuration. This is also what it is used for on `fieldspec`, though it is harder to determine whether it is _also_ used for other things in that case, since `fieldspec` is much more widely used, in many transformers across the codebase. 

While the coupling to the legacy output sort order specifically is unfortunate, in a way, it is not surprising that a resource-oriented tool like Kustomize has other reasons to want to be able to sort by GVK. 

After think about this some more in light of the above, I suggest:
- Copying the relevant code into the new transformer as you said.
- Keeping `Gvk ` sortable here, with a private ordering implementation that we can evolve separately from the legacy order (which can never change) based on the more generic use case.
- Go through this file and update all the comments and method names to ensure they describe deterministic sort ordering justifications and no longer mention the legacy sort transformer and its concerns.